### PR TITLE
Add experimental border colors

### PR DESCRIPTION
**Pull request by kyledurand**
Part of https://github.com/Shopify/polaris-summer-editions/issues/12

### CODE REVIEW

File: polaris-tokens/src/token-groups/color.ts [hunk 1]
```diff
@@ -430,106 +430,132 @@ export const color: {
...
     description: '',
   },
   'color-border-interactive-disabled': {
     value: colors.gray[400],
+    valueExperimental: colorsExperimental.gray[7],
     description: '',
   },
   'color-border-primary': {
     value: colors.green[700],
+    valueExperimental: colorsExperimental.green[8],
```
Comment (laurkim) [inline polaris-tokens/src/token-groups/color.ts]: Looks like `border-primary` maps to `GR08` in the [Figma](https://www.figma.com/file/hL9XRyHGqU8wUtLdOgs0vI/Polaris-Uplift-Library?type=design&node-id=9011%3A154956&t=P03VetQBVKbERO5B-1)

File: polaris-tokens/src/token-groups/color.ts [hunk 2]
```diff
@@ -430,106 +430,132 @@ export const color: {
...
     description: '',
   },
   'color-border-inverse': {
     value: colors.gray[800],
+    valueExperimental: colorsExperimental.gray[13],
     description: '',
   },
   'color-border-strong-hover': {
     value: colors.gray[700],
+    valueExperimental: colorsExperimental.gray[10],
```
Comment (sarahill) [inline polaris-tokens/src/token-groups/color.ts]:
```suggestion
    valueExperimental: colorsExperimental.gray[9],
```

File: polaris-tokens/src/token-groups/color.ts [hunk 3]
```diff
@@ -430,106 +430,132 @@ export const color: {
...
     description: '',
   },
   'color-border-hover': {
     value: colors.gray[600],
+    valueExperimental: colorsExperimental.gray[10],
     description: '',
   },
   'color-border-strong': {
     value: colors.gray[600],
+    valueExperimental: colorsExperimental.gray[9],
```
Comment (sarahill) [inline polaris-tokens/src/token-groups/color.ts]:
```suggestion
    valueExperimental: colorsExperimental.gray[8],
```

File: polaris-tokens/src/token-groups/color.ts [hunk 4]
```diff
@@ -430,106 +430,132 @@ export const color: {
...
     description: '',
   },
   'color-border-disabled': {
     value: colors.gray[400],
+    valueExperimental: colorsExperimental.gray[7],
     description: '',
   },
   'color-border-subdued': {
     value: colors.gray[400],
+    valueExperimental: colorsExperimental.gray[7],
```
Comment (sarahill) [inline polaris-tokens/src/token-groups/color.ts]:
```suggestion
    valueExperimental: colorsExperimental.gray[10],
```

**Review by sarahill — APPROVED** [review]
Made a couple updates and changes are reflected in Figma. Otherwise looks good!

**Pull request merged**